third_party/move/move-vm/test-utils/src/gas_schedule.rs

```diff
@@ -39,6 +35,17 @@ use std::{
     u64,
 };
 
+// gas meter for test-utils
+pub trait TestGasMeter: GasMeter {
+    fn instantiate(&self) -> Self;
```

Comment:
It feels a bit weird to me we are creating a new gas meter from an existing instance. Would we be better served with a `Default::default()` implementation or some sort of factory object that can create gas meters?